CRI-O "grpc: the connection is unavailable" without sudo & "unknown service runtime.RuntimeService" when run with sudo

Hi @estesp 

When i was trying to run your benchmark i got some problems when starting it with or without sudo privileges. 

With sudo: 
```
Error: error during driver info query: rpc error: code = Unimplemented desc = unknown service runtime.RuntimeService
Usage:
  bucketbench run [flags]

Flags:
  -b, --benchmark string   YAML file with benchmark definition
  -h, --help               help for run
  -o, --overhead           Output daemon overhead
  -s, --skip-limit         Skip 'limit' benchmark run
  -t, --trace              Enable per-container tracing during benchmark runs

Global Flags:
      --log-level string   set the logging level (info,warn,err,debug) (default "warn")

ERRO[0029] Command failed with error: error during driver info query: rpc error: code = Unimplemented desc = unknown service runtime.RuntimeService 
```
 
Without sudo: 
```
Error: error during driver info query: rpc error: code = Unavailable desc = grpc: the connection is unavailable
Usage:
  bucketbench run [flags]

Flags:
  -b, --benchmark string   YAML file with benchmark definition
  -h, --help               help for run
  -o, --overhead           Output daemon overhead
  -s, --skip-limit         Skip 'limit' benchmark run
  -t, --trace              Enable per-container tracing during benchmark runs

Global Flags:
      --log-level string   set the logging level (info,warn,err,debug) (default "warn")

ERRO[0028] Command failed with error: error during driver info query: rpc error: code = Unavailable desc = grpc: the connection is unavailable
```

The commands i used: 
`bucketbench --log-level=debug run -b examples/crio.yaml`
`sudo bucketbench --log-level=debug run -b examples/crio.yaml`

Do you have any idea what might be going on/wrong? 

Thank you :)

Hi @estesp,
Thank you for looking into this.
I built using `make binary` and `make install` but I got the same problem I had last time I used this method (might be a different issue? might be myself unable to handle this kind of errors): 
```
go build -ldflags "-X github.com/estesp/bucketbench/cmd.gitCommit="0fc69f1a56f5d9a50faadcebc024d018b9410c89"" -o bucketbench github.com/estesp/bucketbench
can't load package: package github.com/estesp/bucketbench: cannot find package "github.com/estesp/bucketbench" in any of:
	/usr/src/github.com/estesp/bucketbench (from $GOROOT)
	/home/ubuntu/go/src/github.com/estesp/bucketbench (from $GOPATH)
Makefile:15: recipe for target 'binary' failed
make: *** [binary] Error 1
```
Looks like the make binary wants only these specific paths, so I created `home/ubuntu/go/src/github.com/estesp/` directory and cloned the git there.
Now after building I think it can't reach cri-o properly from that location... `sudo crio -version` works, but `sudo ./bucketbench  run -b examples/crio.yaml` gives this:
```
Error: error during driver info query: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /var/run/crio/crio.sock: connect: no such file or directory"
Usage:
  bucketbench run [flags]

Flags:
  -b, --benchmark string   YAML file with benchmark definition
  -h, --help               help for run
  -o, --overhead           Output daemon overhead
  -s, --skip-limit         Skip 'limit' benchmark run
  -t, --trace              Enable per-container tracing during benchmark runs

Global Flags:
      --log-level string   set the logging level (info,warn,err,debug) (default "warn")

ERRO[0032] Command failed with error: error during driver info query: rpc error: code = Unavailable desc = all SubConns are in TransientFailure, latest connection error: connection error: desc = "transport: Error while dialing dial unix /var/run/crio/crio.sock: connect: no such file or directory" 
```
I see it is the same error description you get when running without being root.

--- a/contrib/container_config.json
+++ b/contrib/container_config.json
@@ -4,7 +4,7 @@
 		"attempt": 1
 	},
 	"image": {
-		"image": "alpine:latest"
+		"image": "docker.io/library/alpine:latest"
 	},
 	"command": [
 		"/bin/ls"

--- a/driver/cri.go
+++ b/driver/cri.go
@@ -16,7 +16,7 @@ import (
 )
 
 const (
-	defaultPodImage        = "gcr.io/google_containers/pause:3.0"
+	defaultPodImage        = "k8s.gcr.io/pause:3.1"
 	defaultPodNamePrefix   = "pod"
 	defaultSandboxConfig   = "contrib/sandbox_config.json"
 	defaultContainerConfig = "contrib/container_config.json"

--- a/examples/crio.yaml
+++ b/examples/crio.yaml
@@ -1,5 +1,5 @@
 name: CRICrio
-image: alpine
+image: docker.io/library/alpine:latest
 detached: true
 drivers:
   - 

